### DISCUSSION TITLE
Volume control scroll wheel responsiveness

### DISCUSSION THREAD

**Discussion by Gyoshi** [ppy/osu#23818] [Ideas]
The current control with acceleration is much worse for quick mid-song adjustments compared to stable. I would also argue it is less intuitive/predictable overall.
Related (brief) discussion: https://github.com/ppy/osu/discussions/17418

### Stable:
- First scroll tick activates control, subsequent scrolling changes volume by 5% per tick
- Including the wake-up tick, 2-3 scroll ticks can make significant changes of 5-10% 
<img src="https://github.com/ppy/osu/assets/41743877/b0b5daeb-2af5-4c9a-a833-3e55a8bac357" width="300">

### lazer:
- First scroll tick activates control, second tick 1%, followed by acceleration
- Including the wake-up tick, 2-3 scroll ticks only enacts 1-3% volume change
- It is hard to predict for the user how much of a change they will make with more scroll ticks since both number and speed comes into play, leading to frustration and having to look at the control or listen closely for visual/audio feedback
<img src="https://github.com/ppy/osu/assets/41743877/aa03c2a6-9b7c-491a-a8b1-46616a9b74a7" width="200">

I can only guess that the intention behind the acceleration is to allow for fine volume adjustments (while keeping the whole range accessible). 

Personally I value responsiveness and predictability more than single % levels of precision. As would most osu! players, I imagine—considering the stance on mouse acceleration.

## I propose the following changes:
- Replace with stable behaviour of 5% per tick (with appropriate scaling for precision scroll events—which usually have their own acceleration built in)
- Increase precision to 1% per tick while holding Control / something similar
- Bonus (separate issue/PR): change volume even on the first wake-up scroll tick. It is dubious to enact changes without visual feedback, but even currently it is not immediately obvious that no change is made when initially scrolling. This can be improved by a visual indicator of the initial value. E.g. a differently-coloured segment on the filling circle indicator, or the original numerical value dropping down and fading out similar to the deleted text effect currently implemented.

**Comment by Gyoshi** [comment]
I made a PR mostly to learn about the implementation and make sure I know what I am talking about: https://github.com/ppy/osu/pull/23819

But I am not sure if other people want the acceleration removed, so I will wait and see what people have to say first.

Edit: Closed as premature

**Comment by bdach** [comment]
Rationale for acceleration is given in https://github.com/ppy/osu/pull/12404. What you're proposing is basically a revert of that. However [users still want to be able to apply precise changes](https://github.com/ppy/osu/issues/12403). Your solution for a precise toggle was even proposed but not taken.

I'm not sure I have a horse in this race but to coming out with a PR without knowing all of the above context and basing it entirely on the premise of personal opinion feels pretty premature.

**Reply by Gyoshi** [reply]
Thank you for the relevant links. Please ignore the PR for now (I closed).

I do wonder if the request for finer volume adjustments would have even happened if the volume display had ranged from 0-20.
